README.md

```diff
@@ -42,98 +42,53 @@ cors-proxy stop
 ### CLI configuration
 
 Environment variables:
+
 - `PORT` the port to listen to (if run with `npm start`)
 - `ALLOW_ORIGIN` the value for the 'Access-Control-Allow-Origin' CORS header
 - `INSECURE_HTTP_ORIGINS` comma separated list of origins for which HTTP should be used instead of HTTPS (added to make developing against locally running git servers easier)
 
+## API
 
-## Middleware usage
-
-You can also use the `cors-proxy` as a middleware in your own server.
+A single function is exported that allows a request to be handled:
 
 ```js
-const express = require('express')
-const corsProxy = require('@isomorphic-git/cors-proxy/middleware.js')
-
-const app = express()
-const options = {}
-
-app.use(corsProxy(options))
+import { createServer } from 'node:http';
+import handleRequest from '@isomorphic-git/cors-proxy';
 
-```
-
-### Middleware configuration
-
-*The middleware doesn't use the environment variables.* The options object supports the following properties:
+const server = createServer(handleRequest);
 
-- `origin`: _string_. The value for the 'Access-Control-Allow-Origin' CORS header
-- `insecure_origins`: _string[]_. Array of origins for which HTTP should be used instead of HTTPS (added to make developing against locally running git servers easier)
-- `authorization`: _(req, res, next) => void_. A middleware function you can use to handle custom authorization. Is run after filtering for git-like requests and handling CORS but before the request is proxied.
-
-_Example:_
-```ts
-app.use(
-  corsProxy({
-    authorization: (req: Request, res: Response, next: NextFunction) => {
-      // proxied git HTTP requests already use the Authorization header for git credentials,
-      // so their [Company] credentials are inserted in the X-Authorization header instead.
-      if (getAuthorizedUser(req, 'X-Authorization')) {
-        return next();
-      } else {
-        return res.status(401).send("Unable to authenticate you with [Company]'s git proxy");
-      }
-    },
-  })
-);
-
-// Only requests with a valid JSON Web Token will be proxied
-function getAuthorizedUser(req: Request, header: string = 'Authorization') {
-  const Authorization = req.get(header);
-
-  if (Authorization) {
-    const token = Authorization.replace('Bearer ', '');
-    try {
-      const verifiedToken = verify(token, env.APP_SECRET) as IToken;
-      if (verifiedToken) {
-        return {
-          id: verifiedToken.userId,
-        };
-      }
-    } catch (e) {
-      // noop
-    }
-  }
-}
```

Comment:
We can't just delete existing features. If your PR deleted those, and you can't use them anymore. They will need to be added back, or I will need to revert your PR.